aten/src/ATen/native/EmbeddingBag.cpp

```diff
@@ -433,20 +433,25 @@ std::tuple<Tensor, Tensor, Tensor, Tensor> _embedding_bag_cpu_impl(
     const Tensor& per_sample_weights,
     bool include_last_offset,
     bool requires_grad) {
+  Tensor offsets_ = offsets;
+  if (offsets_.scalar_type() != indices.scalar_type()) {
+    offsets_ = offsets_.toType(indices.scalar_type());
```

Comment:
FYI it's probably better to use type promotion for these conversions (check out `promoteTypes()`), as that avoids potential issues going from e.g. int32 to int8.